### PR TITLE
fix(discover): follow status change after modal closed

### PR DESCRIPTION
**Pull request by MaxtuneLee**
### Description

Use useQuery instead of useMutation to set a optimistic update after "add feed" modal closed, while useQuery is more reasonable since "search" is kind of fetching data not update data.

### Linked Issues
https://github.com/RSSNext/Follow/issues/254

### CODE REVIEW

**Comment by hyoban** [comment]
You should not use `useQuery` directly in the component; see the `defineQuery` and `onSuccess` options in `FeedForm`.

**Comment by MaxtuneLee** [comment]
> You should not use `useQuery` directly in the component; see the `defineQuery` and `onSuccess` options in `FeedForm`.

Thank for the reply, get it

**Comment by Innei** [comment]
One way to look at it is that you don't need to use `useQuery` to drive the data update, you can just use `useSubscriptionByFeedId` hook to get whether the current feed is subscribed or not.

File: src/renderer/src/modules/discover/form.tsx [hunk 1]
```diff
@@ -63,17 +63,6 @@ export function DiscoverForm({ type }: { type: string }) {
       keyword: prefix || "",
     },
   })
-  const mutation = useMutation({
```
Comment (iamvishalkhare) [inline src/renderer/src/modules/discover/form.tsx:66]: Hi there

**Comment by MaxtuneLee** [comment]
> One way to look at it is that you don't need to use `useQuery` to drive the data update, you can just use `useSubscriptionByFeedId` hook to get whether the current feed is subscribed or not.

Thanks for the advice, I separated the Card from DiscoverForm into a separate FeedCard component, and use that hook to get latest status ,which seems to be working well now.

**Pull request closed without merging**